### ISSUE TITLE
As an admin I want to be able to know which user was forbidden (and why)

### ISSUE THREAD

**Issue by php-coder** [openshift/origin#13856]
When user creates a Deployment and pod is getting rejected by one of SCC, many users are trying to grant access to this SCC to the current user.

It's logical that if I'm creating Deployment and I get forbidden error then I don't have some access and I need to grant them to myself. But Kubernetes uses a bit different logic and most of the time, objects are created on behalf of the service account users. Because of this, latter approach won't work and users will be in frustrating state.

Unfortunately ForbiddenError doesn't include information about to whom access were rejected. Because this non-UX behavior most likely won't be fixed, Michal has suggested to include information about forbidden attempts to the audit log. In this case, users will be able to ask an administrator and he will be able to find out which exactly user was forbidden.

CC @mfojtik @soltysh

**Comment by php-coder** [comment]
At this time the only way that I know about is to start server with loglevel 4 and grep for the messages like this:

>I0421 10:35:13.954931   20441 admission.go:88] getting security context constraints for pod  (generate: es-master-2387585559-) with service account info &{system:serviceaccount:myproject:default  [system:serviceaccounts system:serviceaccounts:myproject] map[]}

**Comment by soltysh** [comment]
> Michal has suggested to include information about forbidden attempts to the audit log. In this case, users will be able to ask an administrator and he will be able to find out which exactly user was forbidden.

It make sense from admin point of view to know the reason a particular failure happen. But I don't see every user poking admin to check his particular usecase, I wouldn't want to be that admin. 

What about `oc scc-review` and `oc scc-subject-review` they allow you by pass a resource you want to create and get back information if it gets created or not. Maybe not too specific, but that might be the area we can improve our UX. @sdminonne might know better, he was working on them.

**Comment by php-coder** [comment]
> What about oc scc-review and oc scc-subject-review they allow you by pass a resource you want to create and get back information if it gets created or not

1) it will tell you that you're not allowed but it won't help you to find out how to fix it, make it allowable
2) yesterday it lied to me thay I'm able to create Deployment but later it has failed. I'll report a bug later (or even provide a pull request)

**Comment by soltysh** [comment]
> 1. it will tell you that you're not allowed but it won't help you to find out how to fix it, make it allowable

That's a good place to improve, as I said.

> 2. yesterday it lied to me thay I'm able to create Deployment but later it has failed. I'll report a bug later (or even provide a pull request)

👍 yes please.

**Comment by php-coder** [comment]
> yesterday it lied to me thay I'm able to create Deployment but later it has failed. I'll report a bug later (or even provide a pull request)

Just for the record: I found what was a problem. I used `oc policy scc-subject-review` under `system:admin` user and it showed to me that admin is able to create a deployment but... I should use `oc policy scc-review` instead. The former checks permissions for a user but because deployment is created under service account and not under user it lied to me.
